dokka-subprojects/plugin-base-frontend/src/main/components/search/search.tsx

```diff
@@ -86,33 +83,147 @@ export const WithFuzzySearchFilter = () => {
+    if (inputContainer) {
+      inputContainer.removeEventListener('keydown', handleInputKeyDown);
+    }
+    const clearButton = document.querySelector('[data-test="ring-input-clear"]') as HTMLButtonElement;
+    if (clearButton) {
+      clearButton.removeEventListener('click', handleClearButtonClick);
+    }
+    const closeButton = document.getElementById('search-close-button') as HTMLButtonElement;
+    if (closeButton) {
+      closeButton.removeEventListener('click', handleCloseSearchPopupButtonClick);
+    }
+    if (focusTrap) {
+      focusTrap.destroy();
+      focusTrap = null;
+    }
+    const searchAnchor = document.getElementById('pages-search');
+    if (searchAnchor) {
+      searchAnchor.focus();
+      // Closes the search popup
+      searchAnchor.click();
+    }
   }
 }
 
-function onSearchPopupOpen() {
-  const cleatButton = document.querySelector('[data-test="ring-input-clear"]');
-  if (cleatButton) {
-    const closeButton = document.createElement('button');
-    closeButton.id = 'search-close-button';
-    closeButton.className = 'button button_dropdown button_dropdown_active search--close-button';
-    closeButton.setAttribute('aria-label', 'Close search popup');
-    cleatButton.after(closeButton);
+let activeSearchOptionsObserver: MutationObserver | null = null;
+
+function initActiveSearchOptionObserver() {
+  destroyActiveSearchOptionsObserver();
+
+  activeSearchOptionsObserver = new MutationObserver((mutations) => {
+    mutations.forEach((mutation) => {
+      const searchOptionElement = mutation.target as HTMLElement;
+
+      if (isSearchOptionActive(searchOptionElement) && !isSearchOptionVisible(searchOptionElement)) {
+        searchOptionElement.scrollIntoView({
+          behavior: 'smooth',
+          block: 'center',
+          inline: 'center',
+        });
+      }
+    });
+  });
+
+  const listWrapper = document.querySelector('.ReactVirtualized__Grid');
+
+  if (!listWrapper || !activeSearchOptionsObserver) {
+    return;
   }
-  document.body.style.overflow = 'hidden';
+  activeSearchOptionsObserver.observe(listWrapper, {
+    childList: true,
+    subtree: true,
+    attributes: true,
+    attributeFilter: ['class'],
+  });
 }
 
-function onSearchPopupClose() {
-  document.body.style.overflow = '';
+function destroyActiveSearchOptionsObserver() {
+  if (activeSearchOptionsObserver !== null) {
+    activeSearchOptionsObserver.disconnect();
+    activeSearchOptionsObserver = null;
+  }
+}
+
+function resetSearchOptionsListScrollPosition() {
+  const listWrapper = document.querySelector('.ReactVirtualized__Grid');
+  if (listWrapper) {
+    listWrapper.scrollTop = 0;
+  }
+}
+
+function isSearchOptionActive(element: HTMLElement): boolean {
+  return [...element.classList].some((className) => className.includes('hover'));
+}
+
+function isSearchOptionVisible(element: HTMLElement): boolean {
+  const elementRect = element.getBoundingClientRect();
```

Comment:
`getBoundingClientRect` run reflow could you move it under `if (!listWrapper) return false;`?

Reply:
thanks for noticing! moved